aws-api/src/main/java/com/amplifyframework/api/aws/MutiAuthSubscriptionOperation.java

```diff
@@ -116,7 +116,20 @@ private void dispatchRequest() {
                     MutiAuthSubscriptionOperation.this.subscriptionId = subscriptionId;
                     onSubscriptionStart.accept(subscriptionId);
                 },
-                onNextItem,
+                response -> {
+                    if (response.hasErrors()) {
+                        // If there are auth-related errors, dispatch an ApiAuthException
+                        if (hasAuthRelatedErrors(response)) {
+                            executorService.submit(this::dispatchRequest);
```

Comment:
With this approach, downstream callers (DataStore) are going to get a `ApiAuthException` instead of a `GraphQLResponse` with `errorType=Unauthorized`, right, if using multiauth, right?  I think it would be a little bit better if we could make `MultiAuthSubscriptionOperation` behave more congruently with the regular `SubscriptionOperation`.  

I think this would actually be pretty easy - we would just only call`this::dispatchRequest` here if `authTypes.hasNext()` is true, otherwise, call `onNextItem.accept(response)`

Reply:
I'm testing this locally right now. This would basically serve to short-circuit the last submission to the executor where we would exit anyways because `authTypes.hasNext()` would be false (line 90).

Reply:
I imagine we might need a similar change to any other callers of the `dispatchRequest` method as well.